### PR TITLE
feat(java): Migration from 6.x.x to 7.0.0

### PR DESCRIPTION
**Pull request by romtsn**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

* Migration doc from 6.x.x to 7.0.0
* Added new okhttp integration doc as we can use it in non-android modules now
* Added graphql client errors section to apollo3 as it was missing

### CODE REVIEW

File: src/platforms/java/common/performance/instrumentation/okhttp.mdx
Comment (shanamatthews) [inline src/platforms/java/common/performance/instrumentation/okhttp.mdx:67]:
```suggestion
The SDK uses the `SentryOkHttpInterceptor` to support [Distributed Tracing](/product/sentry-basics/tracing/distributed-tracing/) and creates a span for any HTTP call.
```

File: src/platforms/java/common/performance/instrumentation/okhttp.mdx
Comment (shanamatthews) [inline src/platforms/java/common/performance/instrumentation/okhttp.mdx:68]:
```suggestion
The SDK can give more in-depth information through the `SentryOkHttpEventListener` by creating a span for each operation reported by OkHttp, including DNS setup, proxy selection, HTTP connect, SSL setup, send request headers, send request body, receive response headers, and receive response body.
```

File: src/platforms/java/common/performance/instrumentation/okhttp.mdx
Comment (shanamatthews) [inline src/platforms/java/common/performance/instrumentation/okhttp.mdx:188]: Is this edit correct? Since this is the first time we reference an event listener on the page, I think it's better to specify.

```suggestion
The integration's event listener can propagate calls to another `EventListener` or `EventListener.Factory` that's set on the `SentryOkHttpEventListener` once you create your `OkHttpClient` instance.
```

File: src/platforms/java/common/performance/instrumentation/okhttp.mdx
Comment (shanamatthews) [inline src/platforms/java/common/performance/instrumentation/okhttp.mdx:269]: Which events are "those"? Took my best guess with this edit :)

```suggestion
HTTP client errors sent to Sentry are searchable and you can set alerts on them if you use the `http.url` and `http.status_code` properties. Learn more in our full [Searchable Properties](/product/sentry-basics/search/searchable-properties/) documentation.
```

File: src/platforms/java/migration/6.x-to-7.0.mdx
Comment (shanamatthews) [inline src/platforms/java/migration/6.x-to-7.0.mdx:30]:
```suggestion
  - This can increase your event consumption and may affect your quota, because the SDK will report failed network requests as Sentry events if you're using the `sentry-okhttp` or `sentry-apollo-3` integrations by default. You can customize what errors you want/don't want to have reported for [OkHttp](/platforms/java/performance/instrumentation/okhttp/#http-client-errors) and [Apollo3](/platforms/java/performance/instrumentation/apollo3/#graphql-client-errors) respectively.
```

**Review by shanamatthews — APPROVED** [review]
This looks great! Just a few small suggestions. Please make sure to add in the correct  link to the GitHub project before you merge this, if you accept that suggestion!

**Review by markushi — APPROVED** [review]

**Review by adinauer — APPROVED** [review]
LGTM

**Pull request merged**
